google/cloud/internal/opentelemetry.h

```diff
@@ -66,16 +66,18 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> GetTracer(
  */
 opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpanImpl(
     opentelemetry::nostd::string_view name,
-    opentelemetry::common::KeyValueIterable const& attributes =
-        opentelemetry::common::NoopKeyValueIterable(),
-    opentelemetry::trace::SpanContextKeyValueIterable const& links =
-        opentelemetry::trace::NullSpanContext());
+    opentelemetry::common::KeyValueIterable const& attributes,
+    opentelemetry::trace::SpanContextKeyValueIterable const& links,
+    absl::optional<opentelemetry::trace::StartSpanOptions> const& options =
+        absl::nullopt);
```

Comment:
Is that not the a same as @devbww saying it himself? 😆 

I'm not sure what the alternative would be....

Can you elaborate?

Reply:
The alternative would be to change the (internal-only) callers to pass `nullopt` explicitly instead of having it defaulted.

And then, the "second alternative" would be to have them pass `DefaultStartSpanOptions()` instead of `nullopt`, eliminating the use of `optional`.

Reply:
I think I see what you mean-- emphasis on the not using default arguments for internal interfaces.

Removed the absl::optional and default args for the internal interfaces.